apps/designer/app/designer/features/style-panel/controls/text/text-control.tsx

```diff
@@ -0,0 +1,228 @@
+    inheritedStyle,
+    property: styleConfig.property,
+  });
+
+  if (value === undefined) return null;
+
+  // @todo setValue has a latency issue when updating a value fast(i.e keyboard arrows/scrub control)
+  const setValue = setProperty(styleConfig.property);
+
+  const Icon = iconConfigs[styleConfig.property]?.normal;
+
+  return (
+    <Grid
+      css={{
+        gridTemplateColumns: "$sizes$6 auto $sizes$6",
+        gridTemplateRows: "repeat(1, 1fr)",
+      }}
+    >
+      <Tooltip
+        content={styleConfig.label}
+        delayDuration={200}
+        disableHoverableContent={true}
+      >
+        <IconButton
+          onPointerEnter={({ target }) => {
+            if (value.type !== "unit") return;
+            numericScrubRef.current =
+              numericScrubRef.current ||
+              numericScrubControl(target as HTMLInputElement, {
+                initialValue: value.value as number,
+                onValueChange: ({ value }) => setValue(String(value)),
+              });
+          }}
+          onPointerLeave={() => {
+            if (value.type !== "unit") return;
+            numericScrubRef.current?.disconnectedCallback();
+            numericScrubRef.current = undefined;
+          }}
+          variant="ghost"
+          size="1"
+          css={{
+            ...CSSGridLeft,
+            ...CSSIconButton,
+            ...(isCurrentBreakpoint && {
+              bc: "$colors$blue4",
+              "&:not(:hover)": {
+                color: "$colors$blue11",
+              },
+            }),
+          }}
+        >
+          {Icon && <Icon />}
+        </IconButton>
+      </Tooltip>
+      <TextField
+        css={CSSTextField}
+        spellCheck={false}
+        value={value.value}
+        onChange={(event: ChangeEvent<HTMLInputElement>) =>
+          setValue(event.target.value)
+        }
+        onFocus={(event: FocusEvent<HTMLInputElement>) => event.target.select()}
+        onKeyDown={(event: KeyboardEvent<HTMLInputElement>) => {
+          if (value.type !== "unit") return;
+          if (!["ArrowUp", "ArrowDown"].includes(event.code)) return;
+          event.preventDefault();
+          let currentValue = value.value;
+          let currentDelta = 1;
+          if (event.shiftKey) currentDelta = 10;
+          if (event.altKey) currentDelta = 0.1;
+          if (event.code === "ArrowUp")
+            currentValue = currentValue + currentDelta;
+          if (event.code === "ArrowDown")
+            currentValue = currentValue - currentDelta;
+          const currentValueAsString =
+            currentValue % 1
+              ? currentValue.toPrecision(
+                  Math.abs(currentValue).toString().indexOf(".") + 2
+                )
+              : String(currentValue);
+          setValue(currentValueAsString);
+        }}
+      />
+      {value.type === "unit" ? (
+        <Units
+          value={value.unit}
+          items={units
+            .filter((v) =>
+              [
```

Comment:
why do we need this shortlist instead of just using units entirely?

Reply:
Oh its because you don't want them all in the dropdown. I think it makes sense, but we still need to make other units possible, no?

Reply:
yeah I think this is a good option too, @taylornowotny what do you think? we got a bunch of other units in CSS and even though they are not popular, they exist and not showing them entirely is also strange cherry picking

We can sort these by popularity and they will end up in the list further down and you don't see them if you don't scroll but at least they are there, wdyt?